### PR TITLE
Display test failures in the gradle check status report.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: Daniel (dB.) Doubrovkine <dblock@amazon.com>

### Description

Display test failures in the gradle check status report. 
I tested this by hard-coding a few failed build URIs, so hopefully this doesn't break everything, but to actually open the PR we need a pull request target to work.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@dblock I think we could use community provided test reporters, fe [1], what do you think?

[1] https://github.com/marketplace/actions/test-reporter

**Comment by dblock** [comment]
> @dblock I think we could use community provided test reporters, fe [1], what do you think?
> 
> [1] https://github.com/marketplace/actions/test-reporter

Yeah that's a good idea. Would need to fetch the test results from gradle check to use.

File: .github/workflows/gradle-check.yml [hunk 1]
```diff
@@ -77,6 +77,14 @@ jobs:
               * **URL:** ${{ env.workflow_url }}
               * **CommitID:** ${{ env.pr_from_sha }}
 
+      - name: Extract Test Failure
+        if: ${{ github.event_name == 'pull_request_target' && failure() }}
+        run: |
+            TEST_FAILURES=`curl -s "${{ env.workflow_url }}/testReport/api/json?tree=suites\[cases\[status,className,name\]\]" | jq -r '.. | objects | select(.status=="FAILED",.status=="REGRESSION") | (.className + "." +  .name)' | uniq`
```
Comment (andrross) [inline .github/workflows/gradle-check.yml:83]: Maybe use `uniq -c` to get a count of failures?

**Review by andrross — COMMENTED** [review]
I've tested the curl / jq command and it seems to work. Is there any way to test this end-to-end without just committing it to main?

**Comment by andrross** [comment]
Also I agree there are probably better ways to do this but this seems like a quick and easy improvement that can be done now

File: .github/workflows/gradle-check.yml (hunk 1 above)
Comment (dblock) [reply .github/workflows/gradle-check.yml:83]: I wanted to display the actual failures intead.

**Comment by dblock** [comment]
@andrross This PR is just a lazy feature, you can click on the gradle check and see the test failures. So I don't feel particularly compelled to merge it, but if you think it's useful, maybe you can convince @reta of its benefits? :)

File: .github/workflows/gradle-check.yml (hunk 1 above)
Comment (andrross) [reply .github/workflows/gradle-check.yml:83]: `-c` will give you both the failure and the count:

```
curl -s "https://build.ci.opensearch.org/job/gradle-check/6876/testReport/api/json?tree=suites\[cases\[status,className,name\]\]" | jq -r '.. | objects | select(.status=="FAILED",.status=="REGRESSION") | (.className + "." + .name)' | uniq -c
      2 org.opensearch.index.ShardIndexingPressureConcurrentExecutionTests.testCoordinatingPrimaryThreadedUpdateToShardLimitsAndRejections
      4 org.opensearch.index.mapper.GeoPointFieldMapperTests.testLatLonInArrayMoreThanThreeValues
```

**Comment by reta** [comment]
> @andrross This PR is just a lazy feature, you can click on the gradle check and see the test failures. So I don't feel particularly compelled to merge it, but if you think it's useful, maybe you can convince @reta of its benefits? :)

haha @dblock , I am absolutely not against it, just wanted to make integration of such a feature as simple as possible (using available Github actions but it seems like we need to allow external actions on repository explicitly)

**Comment by andrross** [comment]
The test-reporter action seems to put the results on the github action screen (which you'd have to click to get to) and doesn't seems like it would add much value given that we have a handy link to the test results anyway. Am I misinterpreting that?

I like that this solution ends up being very in-your-face about the flaky failures and you don't have to click anything to see what failed. Might be more motivation to help drive these down :)

**Comment by reta** [comment]
> The test-reporter action seems to put the results on the github action screen (which you'd have to click to get to) and doesn't seems like it would add much value given that we have a handy link to the test results anyway. Am I misinterpreting that?
> 
> I like that this solution ends up being very in-your-face about the flaky failures and you don't have to click anything to see what failed. Might be more motivation to help drive these down :)

@andross fair enough, let's give it a try?

File: .github/workflows/gradle-check.yml (hunk 1 above)
Comment (dblock) [reply .github/workflows/gradle-check.yml]: Added. Nice bash-fu @andrross :)

**Review by andrross — APPROVED** [review]

**Review by kotwanikunal — APPROVED** [review]

**Review by reta — APPROVED** [review]

**Review by mch2 — APPROVED** [review]

**Pull request merged**